### PR TITLE
Group assignment development

### PR DESCRIPTION
**Pull request by Adelina9**

### CODE REVIEW

File: group/assignment-6/MovingCarsSolverTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        public MovingCarsSolverParameterizedTest(int[] desiredCarPositions, int[] currentCarPositions,
+                                                 ArrayList<ArrayList<Integer>> expectedCarMoves) {
+              this.desiredCarPositions = desiredCarPositions;
+              this.currentCarPositions = currentCarPositions;
+              this.expectedCarMoves = expectedCarMoves;
+        }
+
+        private static ArrayList<Move> buildExpectedCarMoves1() {
+            ArrayList<Move> result = new ArrayList<>();
+            result.add(new Move(2, 'B', 'C'));
```
Comment (Adelina9) [inline group/assignment-6/MovingCarsSolverTest.java]: You changed the Move's parameters to int, but still pass characters.

File: group/assignment-6/MovingCarsSolverTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * tests, testing exceptions
+ */
+@RunWith(Enclosed.class)
+public class MovingCarsSolverTest {
+
+    @RunWith(Parameterized.class)
+    public static class MovingCarsSolverParameterizedTest {
+        private int[] desiredCarPositions;
+        private int[] currentCarPositions;
+        private ArrayList<ArrayList<Integer>> expectedCarMoves;
```
Comment (Adelina9) [inline group/assignment-6/MovingCarsSolverTest.java]: Why is expectedCarMoves an ArrayList of ArrayLists?

File: group/assignment-6/MovingCarsSolverTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                new int[]{3, 1, 2, -1},
+                                buildExpectedCarMoves1()
+                            }
+                    });
+        }
+
+        @Test
+        public void testMovingCarsSolver() {
+            ParkingLot parkingLot = new ParkingLot(currentCarPositions, desiredCarPositions);
+            assertArrayEquals(expectedCarMoves.toArray(), MovingCarsSolver.getMoves(parkingLot).toArray());
```
Comment (Adelina9) [inline group/assignment-6/MovingCarsSolverTest.java]: Why use assertArrayEquals and toArray? You need to override equals for the Move class.

File: group/assignment-6/MovingCarsSolverTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.lang.reflect.Array;
+import java.util.*;
```
Comment (Adelina9) [inline group/assignment-6/MovingCarsSolverTest.java]: Wildcard imports should not be used accordint to the google Java Style Guide.

**Comment by monikeu** [comment]
Hi, this branch was for "dirty" development purposes, This pull request is just a misunderstanding in branch  purposes :)

File: group/assignment-6/MovingCarsSolverTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.runners.Parameterized;
+
+/**
+ * Test class having both parameterized tests, easily expandable to have more test cases and single
+ * tests, testing exceptions
+ */
+@RunWith(Enclosed.class)
+public class MovingCarsSolverTest {
+
+    @RunWith(Parameterized.class)
```
[the remark this answers is not in the document]
Comment (monikeu) [reply group/assignment-6/MovingCarsSolverTest.java]: This comment is still actual i think, i noticed when writing that the tests would be more readable when we jad separeated methods, as it is still nice when we have test driven by data, don't know really which approach would be applicable

File: group/assignment-6/MovingCarsSolverTest.java (hunk 4 above)
Comment (monikeu) [reply group/assignment-6/MovingCarsSolverTest.java]: Did not know that! Will use this knowledge

File: group/assignment-6/MovingCarsSolverTest.java (hunk 1 above)
Comment (monikeu) [reply group/assignment-6/MovingCarsSolverTest.java]: Java magic :) More over here https://stackoverflow.com/questions/38355395/why-does-an-int-parameter-allow-passing-a-char-value-but-a-char-parameter-doesn

Comment (Adelina9) [reply group/assignment-6/MovingCarsSolverTest.java]: I know it works too, but I'm not sure why change the move parameteres to int in the first place if you are goint to use characters anyway.

**Pull request merged**